### DISCUSSION TITLE
Sustain control saved as enabled, but not effective when restarting

### DISCUSSION THREAD

**Discussion by maxime-michel** [pedrolcl/VMPK#24] [Q&A] [answered]
Hello,

Thanks for this great tool. I've just tried 0.9.0 hoping https://github.com/pedrolcl/VMPK/issues/18 was describing what I've noticed, but my issue has not been fixed.

I would essentially like the piano to start with sustain on because at times I may play it with the mouse and I would like the note to keep ringing as I switch tools or prepare to record a track.

Sustain is working when I enable it. When I close the app and open it again, I see by the green light that my preference has been saved. However, the notes aren't actually sustained. In order for sustain to really work, I have to disable and enable it again.

Is there a workaround to this? Thanks.

**Comment by pedrolcl** [comment, marked as answer]
This is what the setting "MIDI channel state consistency" is about:

![image](https://github.com/pedrolcl/VMPK/assets/15184708/14d77e22-7c24-4b69-bb98-651c4b367786)

Some synthesizers (for instance, Roland Sound Canvas family) remember the latest controllers and program values for each channel on power on. In that case, this setting is not necessary. For other synthesizers which do not remember the latest state of each MIDI channel, this setting makes VMPK to send the latest values to the synth on startup, trying to keep the UI state and the synth state  in sync.

**Reply by maxime-michel** [reply]
Awesome, thank you @pedrolcl!
